File: kawazoe/q2029/tyama_codeiq2029.kt
fun main(args: Array<String>){
	var s=readLine()
	while(s!=null){
		val n=s.toLong()
		if(n==0L)break
		var r:Long=0
		var i:Long=2
		while(true){
			val x=n-i*(i-1)/2
			if(x<i)break
			if(x%i==0L)r+=x/i
			i+=1
		}
		println(r)
		s=readLine()
	}
}

File: kawazoe/q2029/tyama_codeiq2029.rs
use std::io::BufRead;

fn getline() -> String{
	let mut __ret=String::new();
	std::io::stdin().read_line(&mut __ret).ok();
	return __ret;
}

fn main(){
	//loop {
	//	let s=getline();
	let stdin = std::io::stdin();
	for line in stdin.lock().lines() {
	match line {
        Ok(s) => {
		let n:i64=s.trim().parse().unwrap();
		if n>0 {
		let mut r:i64=0;
		let mut i:i64=2;
		loop {
			let x:i64=n-i*(i-1)/2;
			if x<i {break}
			if x%i==0 {r+=x/i}
			i+=1
		}
		println!("{}",r);
		}
		},
		Err(_) => ()
	}}
}